packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -444,24 +453,41 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
     const userDataToken = Lazy.stringValue({ produce: () => Fn.base64(this.userData.render()) });
     const securityGroupsToken = Lazy.listValue({ produce: () => this.securityGroups.map(sg => sg.securityGroupId) });
 
-    const launchConfig = new CfnLaunchConfiguration(this, 'LaunchConfig', {
-      imageId: imageConfig.imageId,
-      keyName: props.keyName,
-      instanceType: props.instanceType.toString(),
-      securityGroups: securityGroupsToken,
-      iamInstanceProfile: iamProfile.ref,
-      userData: userDataToken,
-      associatePublicIpAddress: props.associatePublicIpAddress,
-      spotPrice: props.spotPrice,
-      blockDeviceMappings: (props.blockDevices !== undefined ?
-        synthesizeBlockDeviceMappings(this, props.blockDevices).map<CfnLaunchConfiguration.BlockDeviceMappingProperty>(
-          ({ deviceName, ebs, virtualName, noDevice }) => ({
-            deviceName, ebs, virtualName, noDevice: noDevice ? true : undefined
-          })
-        ) : undefined),
-    });
+    let launchConfig: CfnLaunchConfiguration | undefined;
+    let launchTemplate: ec2.CfnLaunchTemplate | undefined;
+
+    // create launch template for mixed autoscaling group if mixedInstancesPolicy is defined
+    if (props.mixedInstancesPolicy) {
+      launchTemplate = new ec2.CfnLaunchTemplate(this, "LaunchTemplate", {
+        launchTemplateData: {
+          userData: userDataToken,
+          securityGroupIds: securityGroupsToken,
+          iamInstanceProfile: { arn: iamProfile.attrArn },
+          imageId: imageConfig.imageId,
+          instanceType: props.instanceType.toString(),
+          keyName: props.keyName
+        }
+      });
+    } else {
+      launchConfig = new CfnLaunchConfiguration(this, 'LaunchConfig', {
```

Comment:
The original AutoScalingGroup class was written before we had good class standards and now it's hard to change because of backwards compatibility. But if you're going to accept this as an input parameter (and in general), this class needs an L2 mirror